### PR TITLE
initial inventory implementation

### PR DESCRIPTION
**Pull request by Wong-Innovations**

### CODE REVIEW

File: src/main/java/com/cleanroommc/groovyscript/compat/vanilla/Player.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.cleanroommc.groovyscript.compat.vanilla;
+
+import net.minecraft.item.ItemStack;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Player {
+
+    public static final String givenItems = "groovyscript_given_items";
```
Comment (Rongmario) [inline src/main/java/com/cleanroommc/groovyscript/compat/vanilla/Player.java:10]: `GroovyScript:GivenItems` convention preferred

Comment (Rongmario) [reply src/main/java/com/cleanroommc/groovyscript/compat/vanilla/Player.java:10]: Also, `public static final` variable names should normally be `CONSTANT_CASE`, `GIVEN_ITEMS` here

File: src/main/java/com/cleanroommc/groovyscript/event/EventHandler.java [hunk 2]
```diff
@@ -4,20 +4,51 @@
...
+        }
+        if (!data.getBoolean(Player.givenItems)) {
+            Player.itemMap.forEach((stack, slot) -> {
+                if (slot <= -1) {
+                    ItemHandlerHelper.giveItemToPlayer(event.player, stack.copy());
+                } else {
+                    if (event.player.inventory.getStackInSlot(slot) != ItemStack.EMPTY) {
+                        ItemHandlerHelper.giveItemToPlayer(event.player, stack.copy());
+                    } else {
+                        event.player.inventory.setInventorySlotContents(slot, stack.copy());
```
Comment (Rongmario) [inline src/main/java/com/cleanroommc/groovyscript/event/EventHandler.java:42]: Better would be to use vanilla's `InventoryPlayer#add/addItemStackToInventory` as it handles the `-1` slot already.

```java
Player.itemMap.forEach((stack, slot) -> player.inventory.add(slot, stack.copy())
```

**Review by Rongmario — CHANGES_REQUESTED** [review]

**Comment by Wong-Innovations** [comment]
Should be good to merge now if there are no other suggested changes.

File: src/main/java/com/cleanroommc/groovyscript/event/EventHandler.java [hunk 3]
```diff
@@ -4,20 +4,36 @@
...
 
+    @SubscribeEvent
+    public static void playerLogin(PlayerEvent.PlayerLoggedInEvent event) {
+        NBTTagCompound tag = event.player.getEntityData();
+        NBTTagCompound data = new NBTTagCompound();
+        if (tag.hasKey(EntityPlayer.PERSISTED_NBT_TAG)) {
+            data = tag.getCompoundTag(EntityPlayer.PERSISTED_NBT_TAG);
+        }
+        if (!data.getBoolean(Player.GIVEN_ITEMS)) {
+            Player.ITEM_MAP.forEach((stack, slot) -> event.player.inventory.add(slot, stack.copy()));
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/groovyscript/event/EventHandler.java:31]: I don't really like how this is done.
It might clash with other mods.

**Review by brachy84 — CHANGES_REQUESTED** [review]

**Review by brachy84 — APPROVED** [review]

**Pull request merged**